types/src/contract_event.rs

```diff
@@ -78,6 +79,13 @@ impl ContractEvent {
         ContractEvent::V2(ContractEventV2::new(type_tag, event_data))
     }
 
+    pub fn new_v2_with_type_tag_str(type_tag_str: &str, event_data: Vec<u8>) -> Self {
```

Comment:
is this a test-only function?

Reply:
it's so far only used by a test but I think it should be always-on like `new_v2`.

I'd say it makes slightly more sense than `new_v2`: `new_v2` depends on `TypeTag` which is defined in separate crate...